hibp.js

```diff
@@ -0,0 +1,50 @@
+"use strict";
+
+const got = require("got");
+
+const AppConstants = require("./app-constants");
+const DBUtils = require("./db/utils");
+const pkg = require("./package.json");
+const getSha1 = require("./sha1-utils");
+
+
+const HIBP_USER_AGENT = `${pkg.name}/${pkg.version}`;
+
+
+const HIBP = {
+  async getBreachesForEmail(email) {
+    let foundBreaches = [];
+
+    const sha1 = getSha1(email);
+    const sha1Prefix = sha1.slice(0, 6);
+    const url = `${AppConstants.HIBP_API_ROOT}/breachedaccount/range/${sha1Prefix}`;
+    const headers = {
+      "User-Agent": HIBP_USER_AGENT,
+    };
+
+    console.log(`Fetching ${url}...`);
+
+    try {
+      const response = await got(url, {headers});
+      // Parse response body, format:
+      // {breachedAccount1sha1}:{breach1Name}[,{breach2Name},...]
+      // {breachedAccount2sha1}:{breach3Name}[,{breach4Name},...]
+      for (const breachedAccount of response.body.split("\n")) {
+        if (!breachedAccount) {
+          break;
```

Comment:
Do we want to break here or continue? Not sure what this check is for, actually, when will it be falsey?

Reply:
I saw it falsey after the last line of the response. A newline at the end of the response makes an extra last element of the array which was empty, which was throwing an error when trying to `split` on it.

Reply:
If that's the case, shouldn't we be calling `.trim()` on the `response.body`?

Reply:
Oh yeah ... great point.